### PR TITLE
Fix inheriting from generic @frozen attrs class

### LINKED ISSUE #15658: Mypy errors out with attrs, generics and inheritance
**Bug Report**

When checking python code that uses `attrs` dataclasses, `Generic` classes and inheritance, mypy fails to infer the actual type of a TypeVar.

Swapping `attrs` for std dataclasses removes the error.

Simplifying inheritance chain also removes the error.

I suspect this might be an issue in the mypy-attr plugin.

**To Reproduce**
can't put a playground link since it doesn't support `attrs` sorry.

```python
import abc
from typing import Generic, TypeVar

from attr import frozen


# define a hierarchy of State types
class BaseState(metaclass=abc.ABCMeta):
    ...


T_State = TypeVar("T_State", bound=BaseState, covariant=True)


@frozen()
class JumpState(BaseState):
    start: int


# define a hierarchy of signals


@frozen()
class AbstractBaseSignal(metaclass=abc.ABCMeta):
    pass


T_Signal = TypeVar("T_Signal", bound=AbstractBaseSignal)


@frozen()
class SignalA(Generic[T_State], AbstractBaseSignal, metaclass=abc.ABCMeta):
    state: T_State


@frozen()
class SignalAA(SignalA[T_State], metaclass=abc.ABCMeta):
    pass


# this signal specializes the generic type to a well defined JumpState type
class SignalAAA(SignalAA[JumpState]):
    pass


def foo(s: SignalAAA) -> None:
    reveal_type(s)  # SignalAAA
    reveal_type(s.state)  # T_State instead of JumpState
    reveal_type(s.state.start)  # start is not found on T_State
```

if you swap all the `frozen` decorators for `dataclass` it suddendly works, `s.state` is correctly inferred as a `JumpState`.

if you remove the class `SignalAA` and have `SignalAAA` inherit directly from `SignalA` (i.e. short-circuit the inheritance chain) it works correctly as well.

**Expected Behavior**

I would expect this code to typecheck, especially since it does with standard dataclasses. I suspect a problem with the mypy-attrs plugin.

**Actual Behavior**

mypy output on this example:
```
type_test.py:47: note: Revealed type is "type_test.SignalAAA"
type_test.py:49: note: Revealed type is "T_State`1"
type_test.py:50: error: "T_State" has no attribute "start"  [attr-defined]
type_test.py:50: note: Revealed type is "Any"
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: issue confirmed at least starting from 0.991 to latest version. It was working fine with version 0.910.
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files):
```
[mypy]
python_version = 3.9
warn_unused_configs = True
warn_unused_ignores = True
warn_return_any = True
disallow_untyped_defs = True
disallow_incomplete_defs = True
follow_imports = silent
ignore_missing_imports = True
```
- Python version used: 3.9
- `attrs` package version: 23.1.0 (bug reproduced in 22.1.0 as well)

The bug was observed first while updating our mypy version from 0.910 to 0.991.

### PR DESCRIPTION
**Pull request by ikonst**
Fixes #15658.

### CODE REVIEW

File: mypy/plugins/attrs.py [hunk 1]
```diff
@@ -803,7 +803,7 @@ def _make_frozen(ctx: mypy.plugin.ClassDefContext, attributes: list[Attribute])
         else:
             # This variable belongs to a super class so create new Var so we
             # can modify it.
-            var = Var(attribute.name, ctx.cls.info[attribute.name].type)
+            var = Var(attribute.name, attribute.init_type)
```
Comment (ikonst) [inline mypy/plugins/attrs.py:806]: Actually, this is problematic with converters, e.g.
```python
def converter(s: str) -> int:
    return int(s)

@attrs.define
class C:
    x: int = attrs.field(converter=converter)
```

The `attribute.init_type` will be `str` here :(

Comment (ikonst) [reply mypy/plugins/attrs.py:806]: Ahh, nm, by the time it's deserialized, it's `int` again (there's also converter_init_type).

**Comment by ikonst** [comment]
@hauntsaninja can take a look?

**Review by hauntsaninja — APPROVED** [review]
Not the most familiar with attrs plugin, but this seems reasonable

**Pull request merged**